src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,203 @@
+        testUser.setId(24356876L);
+        testUser.setLogin(validUserLogin);
+        testUser.setPassword("some_valid_password");
+        testUser.setAge(30);
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (!Storage.people.isEmpty()) {
+            Storage.people.clear();
+        }
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_nullUserLogin_NotOk() {
+        testUser.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_nullUserPassword_NotOk() {
+        testUser.setPassword(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_nullUserAge_NotOk() {
+        testUser.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_emptyUserLogin_NotOk() {
+        testUser.setLogin("");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_shortUserLogin_NotOk() {
+        testUser.setLogin("log");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setLogin("login");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_emptyUserPassword_NotOk() {
+        testUser.setPassword("");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_shortUserPassword_NotOk() {
+        testUser.setPassword("pss");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+        testUser.setPassword("short");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_negativeUserAge_NotOk() {
+        testUser.setAge(-20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_noAcceptableUserAge_NotOk() {
+        testUser.setAge(10);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_sameUserLoginRegistered_NotOk() throws CloneNotSupportedException {
+        String repeatingLogin = "same_login_user";
```

Comment:
Save to constant strings or any values that you use not one time

Reply:
changed